### ISSUE TITLE
🐛 BUG: Can't use a component named Head

### ISSUE THREAD

**Issue by robinmetral** [snowpackjs/astro#1172]
### What package manager are you using?

npm

### What operating system are you using?

Mac, Linux

### Describe the Bug

Attempting to create and use an Astro component named `Head` ([like in Next.js](https://nextjs.org/docs/api-reference/next/head)) crashes the dev server with:

```
SyntaxError: [60:303]: Expected ')'
```

I wasn't expecting the component's name to be the issue, so it took me a long time to track down.

Could we:

1. address this behavior, or
2. make the error message more explicit?

Happy to help.

### Steps to Reproduce

Use the attached [CodeSandbox](https://codesandbox.io/s/astro-head-bug-91o68).

As you can see, the index pages uses a `BaseHead` component (like some Astro starters) and the `/bug` page uses the same component, but named `Head` instead.

### Link to Minimal Reproducible Example (Optional)

https://codesandbox.io/s/astro-head-bug-91o68

**Comment by jasikpark** [comment]
I think this is similar to https://github.com/snowpackjs/astro/issues/1133, where astro is having naming collisions between html elements and astro components!

**Comment by robinmetral** [comment]
Ah, could be, thanks for cross-referencing @jasikpark

**Comment by tony-sull** [comment]
Great catch @jasikpark! I think there may actually be a one-off case for `head` in the codebase but that should probably be covered on the fix for #1133 👍

**Comment by natemoo-re** [comment]
This is definitely fixed in `astro@next`! 

The team has decided to close out issues that have been confirmed as fixed by [astro@0.21.0-next.0](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.0), [astro@0.21.0-next.1](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.1), or [astro@0.21.0-next.2](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.2). Our hope is that this will help the [v0.21 milestone](https://github.com/snowpackjs/astro/milestone/1) remain as actionable as possible.

To verify that this issue has been fixed, you may
- Open a new [`astro@next` project on StackBlitz](https://astro.new/minimal@next)
- Locally upgrade to the `astro@next` preview
    ```shell
    npm i astro@next
    # If using Framework renderers...
    npm i @astrojs/renderer-name@next
    ```

If you think we made a mistake, **please reopen this issue**!

If you run into any other problems with `astro@next`, please [open a new issue](https://github.com/snowpackjs/astro/issues/new?assignees=&labels=0.21&template=---01-bug-report.yml&title=%F0%9F%90%9B+BUG%3Aastro@next).
